Modify existing feature: add member details (add)

ryanongra opened the pull request:
- Modified `add` command to take telegram handles instead of address
    - /t tag now recognised as telegram handle
    - /tag used for other tags 
    - automatically adds '@' character is not already preceded by it
- Attempted to modify the validation regex to disallow spaces but to no avail :(, to be done in v1.2b
- Making tags other than the name optional may take more work as the current implementation requires these parameters to be non-null

Closes #33

yongxiangng reviewed (commented):
Understandably, this renaming of address to telegram is rather problematic and I apologise for many repeated comments on the same issue (mostly just to mark and flag the issues so we can check if it has been resolved later on)

Here are some common issues I found:
- [ ] Wild card imports
- [ ] Address book accidentally refactored into telegram book
- [ ] Forgetting to rename address value to telegram handle value. e.g. there are still parts where telegram handles are filled with the String value `"4th street"`
- [ ] Error messages not renamed and the value not changed
- [ ] address.png not updated to telegram.png (also change the picture to updated ui)
- [ ] Several methods name are still not updated and the parameters of certain methods are not updated as well. e.g. setAddress

There might be more things I missed out that I didn't flag out in the review, please be careful and meticulous when refactoring. Once again, I understand this thing is quite a pain and there are many things to change and hard to catch, so it would be best if we all tried to catch the loose ends (5 pairs of eyes are better than 1 pair 👀 👀 👀 👀 👀 )

Please build and run tests before pushing into the master to ensure that you pass the CI. Note the team repo only does checkstyle and tests and doesn't track codecov in the ci.

ryanongra commented:
> Understandably, this renaming of address to telegram is rather problematic and I apologise for many repeated comments on the same issue (mostly just to mark and flag the issues so we can check if it has been resolved later on)
> 
> Here are some common issues I found:
> 
> * [ ]  Wild card imports
> * [ ]  Address book accidentally refactored into telegram book
> * [ ]  Forgetting to rename address value to telegram handle value. e.g. there are still parts where telegram handles are filled with the String value `"4th street"`
> * [ ]  Error messages not renamed and the value not changed
> * [ ]  address.png not updated to telegram.png (also change the picture to updated ui)
> * [ ]  Several methods name are still not updated and the parameters of certain methods are not updated as well. e.g. setAddress
> 
> There might be more things I missed out that I didn't flag out in the review, please be careful and meticulous when refactoring. Once again, I understand this thing is quite a pain and there are many things to change and hard to catch, so it would be best if we all tried to catch the loose ends (5 pairs of eyes are better than 1 pair 👀 👀 👀 👀 👀 )
> 
> Please build and run tests before pushing into the master to ensure that you pass the CI. Note the team repo only does checkstyle and tests and doesn't track codecov in the ci.

@yongxiangng 
Thank you so much for finding all these mistakes 🙏 Sorry for the trouble, I think because I removed some parts when refactoring since a lot of the 'address' keywords are actually part of AddressBook or other libraries which shouldn't be changed, but I removed too many in the end :"). I'll update the PR and run the tests asap.
